CONFIGURATION.md

```diff
@@ -0,0 +1,182 @@
+# Blackbox exporter configuration
+
+The file is written in [YAML format](http://en.wikipedia.org/wiki/YAML), defined by the scheme described below.
+Brackets indicate that a parameter is optional.
+For non-list parameters the value is set to the specified default.
+
+Generic placeholders are defined as follows:
+
+* `<boolean>`: a boolean that can take the values `true` or `false`
+* `<duration>`: a duration matching the regular expression `[0-9]+(ms|[smhdwy])`
+* `<filename>`: a valid path in the current working directory
+* `<string>`: a regular string
+* `<secret>`: a regular string that is a secret, such as a password
+* `<regex>`: a regular expression
+
+The other placeholders are specified separately.
+
+### Module
+```yml
+
+  # The protocol over which the probe will take place (http, tcp, dns, icmp).
+  prober: <prober_string>
+
+  # How long the probe will wait before giving up.
+  [ timeout: <duration> ]
+
+  # At most one of these should be specified.
+  [ http: <http_probe> ]
+
+  [ tcp: <tcp_probe> ]
+
+  [ dns: <dns_probe> ]
+
+  [ icmp: <icmp_probe> ]
+
+```
+
+### http_probe
+```yml
+
+  # Accepted status codes for this probe. Defaults to 2xx.
+  [ valid_status_codes: [ <string>, ... ] | default = "2xx" ]
+
+  # The HTTP method the probe will use.
+  [ method: [ <string> | default = "GET" ] ]
```

Comment:
`[ method:  <string> | default = "GET" ]`

The value isn't optional if the field is specified. Same with others below